src/metabase/api/dataset.clj

```diff
@@ -77,15 +78,45 @@
   (or (get-in ex/export-formats [export-format :context])
       (throw (Exception. (str "Invalid export format: " export-format)))))
 
+(def ^:private date-only-formatter (tformat/formatters :date))
+(def ^:private iso-date-formatter (tformat/formatters :date-time))
+
+(defn- datetime-str->date
```

Comment:
This works, but is pretty inefficient. Since we convert date objects to strings in the query processor, by the time we're here, it's too late. We need to parse the string and re-emit the date. What would be better is leaving the date object alone and instead convert it on the fringes, as we serialize it to JSON (or the export formats).